### ISSUE TITLE
Do not split SQL on delimiters within BEGIN/END blocks for rollback statements

### ISSUE THREAD

**Issue by gallipi** [liquibase/liquibase#3908]
### Search first

- [X] I searched and no similar issues were found

### Description

It would be great if this logic could also apply to the rollback statement. [https://github.com/liquibase/liquibase/pull/1589](https://github.com/liquibase/liquibase/pull/1589)

### Steps To Reproduce

I've tried it, but it fails when detecting the first ";" So,  adding the "rollbackEndDelimiter:/" attribute is needed to solve the problem.

`--changeset liquibase:scheduler_schedule_1 rollbackEndDelimiter:/
--comment: Creation scheduler schedule EVERY_07AM
BEGIN
  SYS.DBMS_SCHEDULER.CREATE_SCHEDULE
    (
      schedule_name    => 'EVERY_07AM'
     ,start_date             => SYSTIMESTAMP
     ,repeat_interval     => 'FREQ=DAILY;INTERVAL=1; BYHOUR=7; BYMINUTE=0;BYSECOND=0'
     ,end_date              => NULL
     ,comments            => 'Scheduler para la ejecucion todos los dias a las 7 de la mañana.'
    );
END;
/

/* liquibase rollback 
BEGIN 
  DBMS_SCHEDULER.DROP_SCHEDULE( 
    schedule_name => 'EVERY_07AM', 
    force => TRUE 
  ); 
END;
/
*/

### Expected/Desired Behavior

Avoid using rollbackEndDelimiter attribute

### Liquibase Version

>=4.15.0

### Database Vendor & Version

any

### Liquibase Integration

CLI

### OS and/or Infrastructure Type/Provider

RedHat

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by nvoxland** [comment]
Yes, that would be a good fix. As our ability to parse sql more smartly than a simple endDelimiter regexp can do so trying to be smarter as a default.

The fix will go into FormattedSqlParser if anyone is interested in picking up the change

**Comment by afn7081** [comment]
I am interested in taking this up if its still open?

**Comment by mamtananavati** [comment]
Is this issue handled ? I can help take this up.

**Comment by filipelautert** [comment]
We're currently triaging our backlog to ensure we can focus on the issues that matter most. With a large number of open issues, it becomes difficult to give each one the attention it deserves.

This issue has been closed due to inactivity — there have been no updates for over 30 days. If this is still relevant to you, please feel free to reopen it or create a new issue with updated context. We appreciate your contribution!
